truss-chains/truss_chains/framework.py

```diff
@@ -723,7 +664,71 @@ def _validate_dependencies(
                 chainlet_cls=marker.chainlet_cls, options=marker.options
             )
             used.add(marker.chainlet_cls)
-        return dependencies
+
+        self.validated_dependencies = dependencies
+
+    def _validate_dependency_param(
+        self, param: inspect.Parameter
+    ) -> Optional[ChainletDependencyMarker]:
+        # Returns `None` if unvalidated.
+        if not self.cls.supports_dependencies():
+            _collect_error(
+                f"Chainlet dependencies are not supported for {self.cls.entity_type()}s, "
+                f"only the optional context argument is allowed.",
+                _ErrorKind.TYPE_ERROR,
+                self._location,
+            )
+            return None
+        # TODO: handle subclasses, unions, optionals, check default value etc.
```

Comment:
Everything here and below is copied over exactly from the previous implementation